cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1717,11 +1770,14 @@ public List<ConfigMap> generatePerBrokerConfigurationConfigMaps(MetricsAndLoggin
                 // script to generate the node configuration.
                 data.put(BROKER_LISTENERS_FILENAME, node.broker() ? listeners.stream().map(ListenersUtils::envVarIdentifier).collect(Collectors.joining(" ")) : null);
 
-                if (useKRaft) {
+                // controller and broker gets the Cluster ID in different states during migration
+                // and they both get it when in full KRaft-mode
+                if (this.kafkaMetadataConfigState.isKRaftInConfiguration(node)) {
```

Comment:
`isKRaftInConfiguration` seems like a super weird state. Also, should it accept oly a boolen flag indicating controller / broker? It does not seem to depend on the broker ID etc.

Reply:
`isKRaftInConfiguration` is not an actual state. It says if, given the current state represented by the `kafkaMetadataConfigState` instance (on which we are calling the method), the KRaft configuration is needed on the node. It's more a shurtcut to check different intervals of states here.
Regarding the flag, are you suggesting a flag because it sounds clearer to you or for any memory performance impact I don't see?

Reply:
In the end I realized that this method was not needed anymore because used just here, so I decided to put the `if` condition directly.